### DISCUSSION TITLE
Can Flask-based Vercel functions have multiple routes / entry points?

### DISCUSSION THREAD

**Discussion by paulkarayan** [vercel/vercel#5084] [Help]
I'm a little confused about the Flask examples provided, namely that it seems like I should be able to do a lot more routing than I can.

things that didn't work for me:
* create a new entry point for each route e.g. dns.py would go to /dns 
I think this is what is suggested here:  https://github.com/vercel/now-examples/issues/163
* create multiple view functions in the same file (e.g. one to /api, one to /dns, etc...)
such as here: https://camillovisini.com/barebone-serverless-flask-rest-api-on-zeit-now/
* mess around with `vercel.json` to try to define `routes` in some clever way
also: https://github.com/vercel/now-examples/issues/163

**what's the best practice and/or set of limitations I'm working with here?**

one thing I'll try is to just do dispatch based on the paths coming in.

```
from flask import Flask, Response
app = Flask(__name__)

@app.route('/', defaults={'path': ''})
@app.route('/<path:path>')
def catch_all(path):
    return Response("<h1>Flask</h1><p>You visited: /%s</p>" % (path), mimetype="text/html")
```

entirely possible that I'm doing it wrong - if so, are there good examples to follow?
or - is this just not well supported? that's fine too - I'm trying to treat these as lightweight middleware rather than something they're not supposed to be. 

thanks!

**Comment by paulkarayan** [comment]
related?
https://stackoverflow.com/questions/62504897/combine-multiple-routes-into-one-serverless-function-in-vercel-using-python

**Comment by paulkarayan** [comment]
also - i cannot get the catchall example to work. even if i delete the other items in the /api directory to avoid nodejs interference (implied in stack overflow above) i am unable to get anything but /api to respond

**Comment by bernatfortet** [comment]
@paulkarayan did you figure it out? If so, would you mind sharing what you did?
